### ISSUE TITLE
sbox.game: Change package's type.

### ISSUE THREAD

**Issue by badandbest** [Facepunch/sbox-issues#5852]
### For?

Other

### What can't you do?

As package types get changed over time, we may feel another type is better suited. But changing a type once it's uploaded is impossible.

In 2023 I made a sprays addon. If I wanted to update this package to make it a library, I'd be out of luck. The best I can do is archive this package and create a new one called `flyestmf.sprays2`. While the ident `flyestmf.sprays` is unusable forever..

![image](https://github.com/user-attachments/assets/65bf955e-915c-46a3-b06c-58a4633ca74a)

### How would you like it to work?

The most obvious solution would be being able to delete packages.

### What have you tried?

* The editor blocks you from uploading an existing package with a different type.
![image](https://github.com/user-attachments/assets/29072aac-e5e0-4c2d-a25d-fd1978a402a3)
* We can't delete a package. It says an archived asset will eventually be deleted. But after three years Luke's ancient, forbbiden org is still not deleted.
![image](https://github.com/user-attachments/assets/7d2d93b9-75da-4cce-93f0-b09c620e5e4b)
![image](https://github.com/user-attachments/assets/58ae8c83-8633-4d50-9b53-5219d4c4e588)
* We can't yet rename an ident or transfer the old package to another org. https://github.com/Facepunch/sbox-issues/issues/5646#issuecomment-2167818015

### Additional context

We have package versions (that we also can't delete). So I don't think games breaking if the type is changed is an unavoidable problem. It might be possible to use an older version of the package from when the game uploaded.

**Comment by DrakeFruit** [comment]
I think being able to change the package type would be nice too

**Comment by QuackCola** [comment]
yeah. i have a map that is the old vpk map package type and i want to replace it with the scene type.
